WalletWasabi/WabiSabi/Client/CoinJoinTracker.cs

```diff
@@ -48,7 +48,10 @@ public class CoinJoinTracker : IDisposable
 	public void Stop()
 	{
 		IsStopped = true;
-		CancellationTokenSource.Cancel();
+		if (!InCriticalCoinJoinState)
```

Comment:
If it's in critical phase then `IsStopped` is set to `true` while the cancellation doesn't happen.

The callers of the `Stop` do are also confused:

- https://github.com/zkSNACKs/WalletWasabi/pull/8140/files#diff-28444f49c77fd0c2cd05f91515914d63fc3c9264ff1f24a56b2fc8e8167df850R208 after calling `Stop` it disposes the `CoinJoinTracker` right away, which disposes the cancellation token source that wasn't cancelled due to critical phase.
- https://github.com/zkSNACKs/WalletWasabi/pull/8140/files#diff-28444f49c77fd0c2cd05f91515914d63fc3c9264ff1f24a56b2fc8e8167df850R232https://github.com/zkSNACKs/WalletWasabi/pull/8140/files#diff-28444f49c77fd0c2cd05f91515914d63fc3c9264ff1f24a56b2fc8e8167df850R232 calls `Stop` which sets `IsStopped` then it checks from the outside the critical phase and it says "It cannot be stopped" even if the `Stop` set the `IsStopped` to true.

Reply:
I will elaborate on this later. IsStopped flag will be used to not restart the coinjoin after it finishes. IsStopped could be renamed to IsStopRequested.